common/scala/src/main/scala/whisk/core/entity/WhiskEntity.scala

```diff
@@ -151,11 +156,39 @@ object WhiskEntityJsonFormat extends RootJsonFormat[WhiskEntity] {
     }
 
     override def write(we: WhiskEntity): JsValue = we match {
-        case a: WhiskAction => WhiskAction.serdes.write(a)
+        case a: WhiskAction     => WhiskAction.serdes.write(a)
         case a: WhiskActivation => WhiskActivation.serdes.write(a)
-        case p: WhiskPackage => WhiskPackage.serdes.write(p)
-        case r: WhiskRule => WhiskRule.serdes.write(r)
-        case t: WhiskTrigger => WhiskTrigger.serdes.write(t)
+        case p: WhiskPackage    => WhiskPackage.serdes.write(p)
+        case r: WhiskRule       => WhiskRule.serdes.write(r)
+        case t: WhiskTrigger    => WhiskTrigger.serdes.write(t)
     }
 }
 
+/**
+ * Trait for the objects we want to size
+ */
+trait ByteSizeable {
+    /**
+     * Method to calculate the size of the object.
+     * The size of the object is defined as the sum of sizes of all parameters, that is stored in the object.
+     *
+     * @return the size of the object as ByteSize
+     */
+    def size(): ByteSize
```

Comment:
`def size: ByteSize` - by convention methods that are idempotent should drop `()`.